src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -2389,190 +2528,233 @@ void GraphOptimizer::MergeTransposeAndReorder(Graph &graph) {
-    //          to the output precision (outPrec)
-    auto mergeTransposeAndReorder = [&](std::shared_ptr<Node>& parentNode, std::shared_ptr<Node>& childNode) {
-        auto parentParentNode = parentNode->getParentEdgeAt(0)->getParent();
-        auto parentParentConstNode = parentNode->getParentEdgeAt(1)->getParent();
-        auto childChildNode = childNode->getChildEdgeAt(0)->getChild();
-
-        auto remEdge = parentNode->getParentEdgeAt(1);
-        graph.RemoveEdge(remEdge);
-
-        // to prevent inPlace conflict we must check that the memory reference is unidirectional or
-        // inPlace memory is not used
-        const auto parentInPlace = parentNode->getParentEdgeAt(0)->inPlace(Edge::LOOK_UP);
-        const auto& childEdges = childNode->getChildEdgesAtPort(0);
-        const auto childInPlace = std::any_of(childEdges.begin(), childEdges.end(),
-            [](const EdgePtr& edge){ return edge->inPlace(Edge::LOOK_DOWN); });
-
-        bool isOptimized = !(parentInPlace && childInPlace);
+        const auto transposeNode = std::dynamic_pointer_cast<Transpose>(parentNode);
+        const auto reorderNode = std::dynamic_pointer_cast<Reorder>(childNode);
+        std::shared_ptr<Reshape> reshapeNode = intermNode != nullptr ? std::dynamic_pointer_cast<Reshape>(intermNode) : nullptr;
+        if (!transposeNode || !reorderNode || (intermNode && !reshapeNode)) {
+            continue;
+        }
 
-        graph.DropNode(parentNode);
-        graph.DropNode(childNode);
+        auto transposeOrder = updateOrder(transposeNode, reshapeNode);
+        auto descBeforeReorder = reorderNode->getParentEdgeAt(0)->getParent()->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc();
+        auto layoutOrder = descBeforeReorder->as<BlockedMemoryDesc>()->getOrder();
 
-        auto inDesc = parentNode->getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].getMemDesc();
-        auto outDesc = childNode->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc();
+        auto inBlockedDesc = reorderNode->getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].getMemDesc()->as<BlockedMemoryDesc>();
+        auto outBlockedDesc = reorderNode->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc()->as<BlockedMemoryDesc>();
+
+        auto& inOrder = inBlockedDesc->getOrder();
+        auto& outOrder = outBlockedDesc->getOrder();
 
-        auto inPrec = inDesc->getPrecision();
-        auto outPrec = outDesc->getPrecision();
+        if (checkAscendingSummaryOrder(transposeOrder, layoutOrder, inOrder, outOrder)) {
+            mergeTransposeReshapeReorder(graph, transposeNode, reshapeNode, reorderNode, false);
+        }
+    }
+}
 
-        auto reorderInDesc = inDesc;
-        auto reorderOutDesc = outDesc->cloneWithNewPrecision(inPrec);
+void GraphOptimizer::MergeReorderAndTranspose(Graph &graph) {
+    auto& graphNodes = graph.GetNodes();
```

Comment:
`MergeReorderAndTranspose` and `MergeTransposeAndReorder` passes look similar, but they are quite different in checks part (please take a look at `updateOrder` lambda for example). So I would prefer to have 2 separate transformations in order to avoid a mess in checks.
The part, which performs the fusion, has already moved to a separate function and reused in both transformations.